### PR TITLE
feature/GE-1115: Persist session when redirecting on domain

### PR DESCRIPTION
**Pull request by AltusBarry**

### CODE REVIEW

File: authentication_service/middleware.py [hunk 1]
```diff
@@ -96,14 +96,18 @@ class ThemeManagementMiddleware(MiddlewareMixin):
     def process_request(self, request):
         if request.path.rstrip("/") in [
                 path.rstrip("/") for path in SESSION_UPDATE_URL_WHITELIST]:
+            current_host = request.get_host()
+            referer = request.META.get("HTTP_REFERER", None)
+            parsed_referer = urlparse(referer)
```
Comment (cobusc) [inline authentication_service/middleware.py]: `urlparse(None)` will raise an exception.

Comment (AltusBarry) [reply authentication_service/middleware.py]: I tested for that before hand. It does not error. The netlock is `urlparse.netloc == ""`

File: authentication_service/middleware.py [hunk 2]
```diff
@@ -96,14 +96,18 @@ class ThemeManagementMiddleware(MiddlewareMixin):
...
+            is_on_domain =  current_host == parsed_referer.netloc
 
             # Grab theme value off of url if available
             query_theme = fetch_theme(request, self.session_theme_key)
             if query_theme:
                 update_session_data(
                     request, self.session_theme_key, query_theme
                 )
-            else:
+            elif not (is_on_domain and referer is not None):
```
Comment (cobusc) [inline authentication_service/middleware.py]:
```
referrer = request.META.get("HTTP_REFERER", None)
if referrer:
    current_host = request.get_host()
    parsed_referrer = urlparse(referrer)
    is_on_domain =  current_host == parsed_referer.netloc
else:
    is_on_domain = False

...
   elif not is_on_domain:
```

File: authentication_service/middleware.py [hunk 3]
```diff
@@ -171,9 +175,13 @@ def process_request(self, request):
...
         if request.path.rstrip("/") in [
                 path.rstrip("/") for path in SESSION_UPDATE_URL_WHITELIST]:
+            current_host = request.get_host()
+            referer = request.META.get("HTTP_REFERER", None)
+            parsed_referer = urlparse(referer)
+            is_on_domain =  current_host == parsed_referer.netloc
 
             # Cleanup session values stored by this middleware
-            if request.method == "GET":
+            if request.method == "GET" and not (is_on_domain and referer is not None):
```
Comment (cobusc) [inline authentication_service/middleware.py]: Similar comments to the ones I did above.

File: authentication_service/middleware.py (hunk 1 above)
Comment (cobusc) [reply authentication_service/middleware.py]: Ah. It raises an exception in Python 2.7, but not 3.6. You can leave it as is.

**Comment by cobusc** [comment]
:+1: You can refactor as discussed.

**Comment by cobusc** [comment]
:+1:

**Pull request closed without merging**